### PR TITLE
Muon - Widget; Home Tab Instrument Widget

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

Requires all core code to be merged.

This PR implements the instrument widget for the home tab of the new muon interface.

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_presenter.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        if selection == "None" or selection == "":
+            self.handle_dead_time_unselected()
+            return
+        try:
+            self._model.check_dead_time_file_selection(selection)
+            self._model.set_user_dead_time_from_ADS(selection)
+            dead_times = self._model.get_dead_time_table().toDict()['dead-time']
+            dead_time_text = self.dead_time_from_data_text(dead_times)
+            self._view.set_dead_time_label(dead_time_text)
+        except ValueError as e:
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_presenter.py]: error

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+    def __init__(self, parent=None):
+        super(InstrumentWidgetView, self).__init__(parent)
+
+        self.layout = QtGui.QGridLayout(self)
+
+        self._button_height = 40
+        self._cached_instrument = ["None", "None"]
+
+        self._button = QtGui.QMessageBox.Ok
+        self._message_box = QtGui.QMessageBox("Error in substitution", "ERROR !", QtGui.QMessageBox.Warning,
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: is there a reason we are not using the muon message box?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+
+    def apply_to_all_hidden(self, hidden=True):
+        if hidden:
+            self.apply_all_label.hide()
+            self.apply_all_checkbox.hide()
+        if not hidden:
+            self.apply_all_label.setVisible(True)
+            self.apply_all_checkbox.setVisible(True)
+
+    def setup_filter_row(self):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: not sure what this does from the name

**Comment by AnthonyLim23** [comment]
is this supposed to have no tests?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: I don't think so no

**Comment by Mantid-Matthew** [comment]
There are no tests in this PR becuase it lacks some of the other files required to test it properly. The relevant tests have been added in #23657

**Review by AnthonyLim23 — DISMISSED** [review]

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+        self.apply_all_checkbox = QtGui.QCheckBox(self)
+
+        self.horizontal_layout_6 = QtGui.QHBoxLayout()
+        self.horizontal_layout_6.setObjectName("horizontalLayout6")
+        self.horizontal_layout_6.addWidget(self.apply_all_label)
+        self.horizontal_layout_6.addWidget(self.apply_all_checkbox)
+
+    def setup_interface(self):
+        self.setObjectName("InstrumentWidget")
+        # self.resize(500, 100)
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: Commented-out code?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+        self.vertical_layout.addItem(self.horizontal_layout)
+        self.vertical_layout.addItem(self.horizontal_layout_2)
+        self.vertical_layout.addItem(self.horizontal_layout_3)
+        self.vertical_layout.addItem(self.horizontal_layout_4)
+        self.vertical_layout.addItem(self.dead_time_file_layout)
+        # self.vertical_layout.addItem(self.horizontal_layout_5)
+        self.vertical_layout.addItem(self.horizontal_layout_6)
+
+        self.group.setLayout(self.layout)
+        # self.group.setLayout(self.vertical_layout)
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: Commented-out code? Above as well.

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+        self.timezero_checkbox.setObjectName("timeZeroCheckbox")
+        self.timezero_checkbox.setChecked(True)
+
+        self.timezero_label_2 = QtGui.QLabel(self)
+        self.timezero_label_2.setObjectName("timeZeroLabel")
+        self.timezero_label_2.setText(" )")
+
+        self.horizontal_layout_2 = QtGui.QHBoxLayout()
+        self.horizontal_layout_2.setObjectName("horizontalLayout2")
+        # self.horizontal_layout_2.addWidget(self.timezero_edit)
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: Commented-out code?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 7]
```diff
@@ -0,0 +1,587 @@
...
+
+        self.firstgooddata_edit = QtGui.QLineEdit(self)
+        firstgooddata_validator = QtGui.QRegExpValidator(QtCore.QRegExp(valid_float_regex), self.timezero_edit)
+        self.timezero_edit.setValidator(firstgooddata_validator)
+        self.firstgooddata_edit.setObjectName("firstgooddataEdit")
+        self.firstgooddata_edit.setText("")
+
+        self.firstgooddata_unit_label = QtGui.QLabel(self)
+        self.firstgooddata_unit_label.setObjectName("firstgooddataUnitLabel")
+        self.firstgooddata_unit_label.setText(u" µs (From data file ")  # "micro seconds ( ")
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: The comment looks a bit strange. The label text as well.

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 8]
```diff
@@ -0,0 +1,587 @@
...
+        self.firstgooddata_checkbox.setObjectName("firstgooddataCheckbox")
+        self.firstgooddata_checkbox.setChecked(True)
+
+        self.firstgooddata_label_2 = QtGui.QLabel(self)
+        self.firstgooddata_label_2.setObjectName("timeZeroLabel")
+        self.firstgooddata_label_2.setText(" )")
+
+        self.horizontal_layout_3 = QtGui.QHBoxLayout()
+        self.horizontal_layout_3.setObjectName("horizontalLayout3")
+        # self.horizontal_layout_3.addWidget(self.firstgooddata_edit)
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: Commented-out code?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py [hunk 9]
```diff
@@ -0,0 +1,587 @@
...
+        self.horizontal_layout_5.addWidget(self.rebin_help_button)
+
+        self.rebin_steps_label.hide()
+        self.rebin_steps_edit.hide()
+        self.rebin_variable_label.hide()
+        self.rebin_variable_edit.hide()
+
+        # self.layout.addWidget(self.rebin_label, 5, 0)
+        # self.layout.addWidget(self.rebin_selector, 5, 1)
+        # self.layout.addItem(self.horizontal_layout_5, 5, 2)
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_view.py]: Commented-out code?

File: scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_presenter.py [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+        self._model.add_variable_binning(variable_bin_size)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Instrument
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_instrument_changed(self):
+        """
+        User changes the selected instrument.
+        """
```
Comment (soininen) [inline scripts/Muon/GUI/Common/home_instrument_widget/home_instrument_widget_presenter.py]: Single line docstrings should be one-liners, see [here](https://www.python.org/dev/peps/pep-0257/).

**Review by soininen — CHANGES_REQUESTED** [review]
A few minor issues below.

**Review by soininen — APPROVED** [review]

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**